### PR TITLE
Deprecated OCP interface to fetch background job type

### PR DESCRIPTION
**Pull request by MorrisJobke**
* was not used by apps and also is not needed
* migrated the documentation to IJobList

### CODE REVIEW

File: cron.php [hunk 1]
```diff
@@ -102,8 +102,8 @@
 		}
 
 		// We call ownCloud from the CLI (aka cron)
```
Comment (nickvergessen) [inline cron.php]: Fix this too

**Review by nickvergessen — APPROVED** [review]

File: lib/public/BackgroundJob/IJobList.php [hunk 2]
```diff
@@ -29,6 +29,19 @@
 /**
  * Interface IJobList
  *
+ * This interface provides functions to register background jobs
+ *
+ * To create a new background job create a new class that inherits from either
+ * \OC\BackgroundJob\Job, \OC\BackgroundJob\QueuedJob or
```
Comment (rullzer) [inline lib/public/BackgroundJob/IJobList.php]: Not for the scope of this PR. But we really should move interfaces for that to OCP.

**Review by rullzer — APPROVED** [review]

File: lib/public/BackgroundJob/IJobList.php (hunk 2 above)
Comment (nickvergessen) [reply lib/public/BackgroundJob/IJobList.php]: And abstract classes

**Comment by MorrisJobke** [comment]
I pushed the fixed doc comments -> merge

**Pull request merged**